# Задание №3. Решение задачи условной выпуклой оптимизации с помощью метода множителей Лагранжа

**Ф.И.О.:** Филиппов Владимир Леонидович    
**Поток:** МетОпт 1.2  
**Вариант:** 6 

Задача:
- минимизировать $f(x,y)=x^2+3y^2$
- при $g_1=x^2+y^2-1=0$ и $g_2=(x-1)^2+(y-1)^2-2=0$.

In [1]:
import math

sqrt7 = math.sqrt(7)

# две допустимые точки (получены из пересечения окружности x^2+y^2=1 и прямой x+y=1/2)
x1 = (1 + sqrt7) / 4
y1 = (1 - sqrt7) / 4

x2 = (1 - sqrt7) / 4
y2 = (1 + sqrt7) / 4

def f(x, y):
    return x*x + 3*y*y

def g1(x, y):
    return x*x + y*y - 1

def g2(x, y):
    return (x-1)**2 + (y-1)**2 - 2

print("P1 =", (x1, y1))
print("P2 =", (x2, y2))

print("\nЗначения функции:")
print("f(P1) =", f(x1, y1))
print("f(P2) =", f(x2, y2))

print("\nПроверка ограничений (должно быть около 0):")
print("g1(P1), g2(P1) =", g1(x1, y1), g2(x1, y1))
print("g1(P2), g2(P2) =", g1(x2, y2), g2(x2, y2))


P1 = (0.9114378277661477, -0.4114378277661477)
P2 = (-0.4114378277661477, 0.9114378277661477)

Значения функции:
f(P1) = 1.3385621722338525
f(P2) = 2.661437827766148

Проверка ограничений (должно быть около 0):
g1(P1), g2(P1) = 0.0 0.0
g1(P2), g2(P2) = 0.0 0.0


## Итог
Минимум достигается в точке $\left(\frac{1+\sqrt7}{4},\frac{1-\sqrt7}{4}\right)$ и равен примерно **1.338562**.